java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -221,19 +227,121 @@ private static Consumer createConsumer(
     return consumer;
   }
 
+  private static Consumer createSkipConsumer(Schema schema) {
+
+    SkipFunction skipFunction;
+    Type type = schema.getType();
+
+    switch (type) {
+      case UNION:
+        int size = schema.getTypes().size();
+        long nullCount = schema.getTypes().stream().filter(s -> s.getType() == Type.NULL).count();
+        if (size == 1) {
+          return createSkipConsumer(schema.getTypes().get(0));
```

Comment:
can you add a unit test for unions and arrays to make sure they are skipped correctly?  I  I think you somehow need to read off the first union value as well.  And it is a good idea to check arrays since there was a bug their. initially. Generally, any time a bug is found in review it is a good idea to add a test case for it.

Reply:
>I I think you somehow need to read off the first union value as well
You are right, thanks for catching this.